core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -99,6 +102,16 @@ public function reset(): void
     {
         $this->adapterCache = [];
         $this->isFrontend = false;
+
+        if (!$this->isInitialized()) {
+            return;
+        }
+
+        Environment::reset();
+        Input::resetCache();
+        Input::resetUnusedGet();
+        InsertTags::reset();
+        Registry::getInstance()->reset();
```

Comment:
It should be possible to call `Registry::reset()` here, shouldn't it? This would require to make the `Registry` properties static as it is in the other singleton classes mentioned above. @contao/developers WDYT?

Reply:
How about https://github.com/contao/contao/pull/1182#discussion_r390628098 ?

Reply:
IDK. What is the problem with making the `Registry` properties static? Should have some performance benefits as well.

Reply:
It would mean a complete rewrite of the class and `getInstance()` would be meaningless then. I don’t think that static properties would be better in terms of performance.

Reply:
It's just that the `Input` and `Environment` classes work exactly this way. Maybe the properties of these classes should not be static either then?

Reply:
But you cannot really change static/nonstatic. This will trigger warnings for people using `$instance->method()` instead of `Class::$method` (see https://www.php.net/manual/en/language.oop5.static.php). The only way around this would imo be to delegate all calls obtained via `getInstance()` to their static counterpart. But I don't see the point of doing this.